sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -224,7 +224,7 @@ public virtual TokenCredential Credential
                                 GetVariable("TENANT_ID"),
                                 GetVariable("CLIENT_ID"),
                                 clientSecret,
-                                new ClientSecretCredentialOptions { AuthorityHost = new Uri(AuthorityHostUrl) });
+                                new ClientSecretCredentialOptions { AuthorityHost = new Uri(GetVariable("AZURE_AUTHORITY_HOST")) });
```

Comment:
why not add this as a fallback [here](https://github.com/Azure/azure-sdk-for-net/pull/41734/files#diff-2542987e61fcb914812b67707e45371fcbfb04d0ffeff71d7a5335d71c6e3b4fR177)?

Reply:
GetRecordedOptionalVariable throws if there is no recording.